handlers/api.py

```diff
@@ -434,6 +434,20 @@ def get(self):
         return self.write(obj)
 
 
+class ShakesHandler(BaseHandler):
+
+    def check_xsrf_cookie(self):
+        return
+
+    @oauth2authenticated
+    def get(self, type, resource=''):
+        if type == 'shake_path':
+            s = Shake.get('path=%s and deleted=0', resource)
+        elif type == 'shake_id':
+            s = Shake.get('id=%s and deleted=0', resource)
+        return self.write(s.as_json(extended=True))
```

Comment:
Good practice might be avoiding `s` being undefined here (tho your path checks are good).
Possibly have s declared as a default {} object or even two separate routes?

Reply:
Here's where I'll admit that I don't actually know Python at all. I just copied the UserHandler class. I'm happy to make any changes. Is this what you meant?

```py
s = {}
if type == 'shake_path':
    s = Shake.get('path=%s and deleted=0', resource)
elif type == 'shake_id':
    s = Shake.get('id=%s and deleted=0', resource)
return self.write(s.as_json(extended=True))
```